app/javascript/components/visual-settings-form/index.jsx

```diff
@@ -15,6 +15,7 @@ const VisualSettingsForm = ({ recordId }) => {
   }, [recordId]);
 
   const onSubmit = (settings) => {
+    settings.display.timezone = settings.display.timezone.value ? settings.display.timezone.value : settings.display.timezone;
```

Comment:
Can you add a `// HACK:` comment here pointing to the issue that you opened?  I presume that once DDF is fixed, this line is no longer needed, so it would help during a followup so we can remove this line.

Reply:
Actually, that line has nothing to do with the DDF issue. Basically, right now timezone gets stored as an object like this
`timezone: { label: "(GMT-05:00) Eastern Time (US & Canada)", value: "Eastern Time (US & Canada)" }` when it's expected to just be stored as just the value i.e. `timezone: "Eastern Time (US & Canada)"`. Adding this line seemed like the better solution then looking for everywhere timezone is used and changing the code to look for something like `timezone.value` instead.

Reply:
This issue was introduced when `ComboBox` was used for `isSearchable` variant (Carbon's `Select` cannot filter), that behaves differently than the `Select` - specifically, it handles values differently, so that's the reason you see wrong initial value and you had to pick the `value` property out of the timezone. Both of these are valid bugs and should not happen. However, this workaround and the workaround I posted in the issue should work well until a fix is made. (Do you think you are going to be able to fix it in DDF? 🤔  Unfortunately, DDF team has little resources now to fix it by ourselves as it is not a critical issue.)

Reply:
Oh, @Hyperkid123 opened a PR: https://github.com/data-driven-forms/react-forms/pull/1023